src/ncp-spinel/SpinelNCPTaskChangeNetData.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ *
+ * Copyright (c) 2016 Nest Labs, Inc.
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#if HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include "assert-macros.h"
+#include <syslog.h>
+#include <errno.h>
+#include "SpinelNCPTaskChangeNetData.h"
+#include "SpinelNCPInstance.h"
+#include "spinel-extra.h"
+
+using namespace nl;
+using namespace nl::wpantund;
+
+nl::wpantund::SpinelNCPTaskChangeNetData::SpinelNCPTaskChangeNetData(
+	SpinelNCPInstance* instance,
+	CallbackWithStatusArg1 cb,
+	const Data& change_net_data_command,
+	int timeout
+):	SpinelNCPTask(instance, cb), mChangeNetDataCommand(change_net_data_command)
+{
+	mNextCommandTimeout = timeout;
+}
+
+int
+nl::wpantund::SpinelNCPTaskChangeNetData::vprocess_event(int event, va_list args)
+{
+	int ret = kWPANTUNDStatus_Failure;
+
+	EH_BEGIN();
+
+	// The first event to a task is EVENT_STARTING_TASK. The following
+	// line makes sure that we don't start processing this task
+	// until it is properly scheduled. All tasks immediately receive
+	// the initial `EVENT_STARTING_TASK` event, but further events
+	// will only be received by that task once it is that task's turn
+	// to execute.
+	EH_WAIT_UNTIL(EVENT_STARTING_TASK != event);
+
+	mNextCommand = SpinelPackData(
+		SPINEL_FRAME_PACK_CMD_PROP_VALUE_SET(SPINEL_DATATYPE_BOOL_S),
+		SPINEL_PROP_THREAD_ALLOW_LOCAL_NET_DATA_CHANGE,
+		true
+	);
+
+	EH_SPAWN(&mSubPT, vprocess_send_command(event, args));
+
+	ret = mNextCommandRet;
+
+	require_noerr(ret, on_error);
+
+	mNextCommand = mChangeNetDataCommand;
+
+	EH_SPAWN(&mSubPT, vprocess_send_command(event, args));
+
+	ret = mNextCommandRet;
+
+	require_noerr(ret, on_error);
```

Comment:
If this fails we need to make sure we set `PROP_THREAD_ALLOW_LOCAL_NET_DATA_CHANGE` back to `false`, otherwise we won't be able to make any more changes.